packages/flutter/test/material/raised_button_test.dart

```diff
@@ -617,6 +617,45 @@ void main() {
     expect(box.size, equals(const Size(76, 36)));
     expect(childRect, equals(const Rect.fromLTRB(372.0, 293.0, 428.0, 307.0)));
   });
+
+  testWidgets('RaisedButton.icon responds to applied padding', (WidgetTester tester) async {
+    const Key buttonKey = Key('test');
+    const Key labelKey = Key('label');
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: Center(
+            child: RaisedButton.icon(
+              icon: const Icon(Icons.add),
+              padding: const EdgeInsets.all(16),
+              key: buttonKey,
+              onPressed: () {},
+              label: const Text(
+                'Hello',
+                key: labelKey,
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final Rect paddingRect = tester.getRect(find.byType(Padding));
+    final Rect labelRect = tester.getRect(find.byKey(labelKey));
+    final Rect iconRect = tester.getRect(find.byType(Icon));
+
+    // Check the position of Text and Icon
+    expect(paddingRect.right, labelRect.right + 16);
+    expect(paddingRect.left, iconRect.left - 16);
+    // Check which is the taller widget from Icon and Text and use that
+    // to check the top and bottom edges
```

Comment:
nit: slight rewording
```suggestion
    // Use the taller widget to check the top and bottom padding.
```